### ISSUE TITLE
Regarding the output images of lr0, lr1 and lr2

### ISSUE THREAD

**Issue by lzqwq** [ultralytics/yolov5#13547]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have run the experiment many times and outputted the images of lr0, lr1, lr2. I noticed that the data inside is not random. Is it possible to set it? The core problem I am facing now is that the previous experiments I ran with lr0, lr1, lr2 were relatively high, which is represented by the solid line in the figure, and the experimental effect was also better. However, recently the experiments have deteriorated. No matter how I modify the model, lr0, lr1, lr2 remain the same, which is represented by the dashed line in the figure.
I would be very grateful if you could address my question.

![Image](https://github.com/user-attachments/assets/71eb5812-4fd9-49db-991f-bb195b9ed4d9)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lzqwq, thank you for reaching out and sharing your question about YOLOv5 🚀! Your detailed description and attached image are helpful in understanding the issue.

If this is a 🐛 Bug Report, could you please provide a **minimum reproducible example (MRE)**? This will help us better investigate the issue. Include the exact commands or scripts you used, along with any specific configurations or customizations.

If this is a ❓ Question about custom training or hyperparameters, please share additional details such as:
- Training logs or metrics
- Dataset examples
- Steps you've taken to modify the model or address the issue
- Confirmation that you've reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

## Requirements

Ensure you're using [**Python>=3.8.0**](https://www.python.org/) and have installed all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Verified Environments

YOLOv5 runs smoothly in these environments (pre-installed with [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/)):

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests ensure YOLOv5's functionality for training, validation, inference, export, and benchmarking across macOS, Windows, and Ubuntu.

This is an automated response to guide you as quickly as possible 😊. An Ultralytics engineer will review your issue and assist you soon. Thank you for your patience and for using YOLOv5! 🚀

**Comment by pderrenger** [comment]
@lzqwq the learning rate plots (lr0, lr1, lr2) in YOLOv5 training outputs are determined by the learning rate scheduler and hyperparameters in your `hyp.yaml` file. These curves follow a deterministic cosine annealing schedule with warmup, not random values. To modify them:

1. Adjust the base learning rate (`lr0`) in your hyperparameter configuration
2. Modify the scheduler parameters like `lrf` (final learning rate factor) in `train.py`

If you're seeing inconsistent behavior between experiments:
- Verify you're using identical hyperparameter files for comparable runs
- Ensure you're training with the same dataset versions
- Confirm package versions match between experiments (`pip list | grep ultralytics`)

For detailed guidance on hyperparameter tuning, see the YOLOv5 Hyperparameter Evolution documentation.

**Comment by lzqwq** [comment]
> [@lzqwq](https://github.com/lzqwq) the learning rate plots (lr0, lr1, lr2) in YOLOv5 training outputs are determined by the learning rate scheduler and hyperparameters in your `hyp.yaml` file. These curves follow a deterministic cosine annealing schedule with warmup, not random values. To modify them:
> 
> 1. Adjust the base learning rate (`lr0`) in your hyperparameter configuration
> 2. Modify the scheduler parameters like `lrf` (final learning rate factor) in `train.py`
> 
> If you're seeing inconsistent behavior between experiments:
> 
> * Verify you're using identical hyperparameter files for comparable runs
> * Ensure you're training with the same dataset versions
> * Confirm package versions match between experiments (`pip list | grep ultralytics`)
> 
> For detailed guidance on hyperparameter tuning, see the YOLOv5 Hyperparameter Evolution documentation.

OK. Thank you for your reply.

**Comment by pderrenger** [comment]
@lzqwq for persistent learning rate behavior across experiments despite hyperparameter changes, consider trying **Hyperparameter Evolution** to systematically optimize `lr0` and `lrf`. This automated approach often resolves inconsistencies better than manual adjustments. You can implement it by adding `--evolve` to your training command and following the [Hyperparameter Evolution Documentation](https://github.com/ultralytics/yolov5#hyperparameter-evolution). 

Also, ensure you're not reusing previous training logs (delete `runs/` folder between tests if needed). Let us know if evolution yields better results! 🚀
